src/run.ts

```diff
@@ -8,11 +8,15 @@ import {
   BulkArtifact,
   BulkArtifactMimeType,
   updateBulkArtifactsStatus,
+  BulkArtifactKind,
   BulkArtifactStatus
 } from './api/captain'
-import {getInputs, mimeTypeFromExtension, InputArtifact} from './utils'
+import {getInputs, mimeTypeFromExtension} from './utils'
 
-type Artifact = InputArtifact & {
```

Comment:
I didn't know if there was a clever way to do intersection types with dropping `save_as` from `InputArtifact`, so I just manually copied `kind` and `path` from `InputArtifact` to `Artifact`

Reply:
You can do `Omit<InputArtifact, 'save_as'> & { mime_type: ... }`.

Reply:
😮

Reply:
`Omit` and `Pick` are two of the coolest utilities in TS. Actually everything on https://www.typescriptlang.org/docs/handbook/utility-types.html is crazy powerful.